packages/frontend/app/components/curriculum-inventory/new-sequence-block.js

```diff
@@ -45,56 +46,84 @@ export default class CurriculumInventoryNewSequenceBlock extends Component {
   @NotBlank()
   @AfterDate('startDate', { granularity: 'day' })
   endDate;
-  @tracked orderInSequence = 0;
-  @tracked linkableCourses = [];
+  @tracked orderInSequence = null;
   @tracked
   @NotBlank()
   @IsInt()
   @Gte(0)
   @Custom('validateMaximumCallback', 'validateMaximumMessageCallback')
   maximum = 0;
   @tracked @NotBlank() @IsInt() @Gte(0) minimum = 0;
-  @tracked orderInSequenceOptions = [];
   @tracked required;
-  @tracked requiredOptions = [];
   @tracked @NotBlank() @Length(1, 200) title;
   @tracked track = false;
 
   constructor() {
     super(...arguments);
-    this.childSequenceOrderOptions = [
+    this.childSequenceOrder = this.childSequenceOrderOptions[0];
+    this.required = this.requiredOptions[0];
+  }
+
+  get childSequenceOrderOptions() {
+    return [
       { id: '1', title: this.intl.t('general.ordered') },
       { id: '2', title: this.intl.t('general.unordered') },
       { id: '3', title: this.intl.t('general.parallel') },
     ];
-    this.requiredOptions = [
+  }
+
+  get requiredOptions() {
+    return [
       { id: '1', title: this.intl.t('general.required') },
       { id: '2', title: this.intl.t('general.optionalElective') },
       { id: '3', title: this.intl.t('general.requiredInTrack') },
     ];
   }
 
-  get linkedCourseIsClerkship() {
-    if (!this.course) {
-      return false;
+  @cached
+  get defaultOrderInSequence() {
+    if (!this.isInOrderedSequence || !this.args.parent) {
+      return 0;
     }
-    return !!this.course.belongsTo('clerkshipType').id();
+    return 1;
   }
 
-  get hasZeroDuration() {
-    const num = Number(this.duration);
-    if (Number.isNaN(num)) {
-      return false;
+  @cached
+  get orderInSequenceOptionsData() {
+    return new TrackedAsyncData(
+      this.getOrderInSequenceOptions(this.isInOrderedSequence, this.args.parent),
+    );
+  }
+
+  get orderInSequenceOptions() {
+    return this.orderInSequenceOptionsData.isResolved ? this.orderInSequenceOptionsData.value : [];
+  }
+
+  async getOrderInSequenceOptions(isInOrderedSequence, parent) {
+    const rhett = [];
+    if (!isInOrderedSequence || !parent) {
+      return rhett;
     }
-    return 0 === num;
+    const siblings = await parent.children;
```

Comment:
Because this async is hidden inside this method I'm not sure if `orderInSequenceOptionsData` will recompute if `parent.children` is updated. Probably better to create a `childrenData` and then pass that into a getter.

Reply:
refactored out into `siblings` getter. :white_check_mark: